barretenberg/cpp/src/barretenberg/bb/readme.md

```diff
@@ -52,20 +52,84 @@ Certain `bb` commands will expect the tool `jq` to already be installed. If `jq
 TODO: https://github.com/AztecProtocol/aztec-packages/issues/7511
 
 For quick reference:
+- Noir v0.34.0 <> BB v0.55.0
+- Noir v0.33.0 <> BB v0.47.1
 - Noir v0.32.0 <> BB v0.46.1
 - Noir v0.31.0 <> BB v0.41.0
 
 ### Usage
 
 TODO: https://github.com/AztecProtocol/aztec-packages/issues/7600
 
-Full list of available commands:
-https://github.com/AztecProtocol/aztec-packages/blob/1a97698071a667cd56510c7b7201373a9ac9c646/barretenberg/cpp/src/barretenberg/bb/main.cpp#L1361-L1493
+All available `bb` commands:
+https://github.com/AztecProtocol/aztec-packages/blob/barretenberg-v0.55.0/barretenberg/cpp/src/barretenberg/bb/main.cpp#L1369-L1512
 
 #### FilePath vs Stdout
 
 For commands which allow you to send the output to a file using `-o {filePath}`, there is also the option to send the output to stdout by using `-o -`.
 
+#### Usage with UltraHonk
+
+Documented with Noir v0.33.0 <> BB v0.47.1:
+
+##### Proving and verifying
+
+1. Follow [the Noir docs](https://noir-lang.org/docs/getting_started/hello_noir/) to compile and generate witness of your Noir program
+
+2. Prove the valid execution of your Noir program running:
+
+    ```bash
+    bb prove_ultra_honk -b ./target/hello_world.json -w ./target/witness-name.gz -o ./target/proof
+    ```
+
+3. Compute the verification key for your Noir program running:
+
+    ```bash
+    bb write_vk_ultra_honk -b ./target/hello_world.json -o ./target/vk
+    ```
+
+4. Verify your proof running:
+
+    ```bash
+    bb verify_ultra_honk -k ./target/vk -p ./target/proof
+    ```
+
+    If successful, the verification will complete in silence; if unsuccessful, the command will trigger logging of the corresponding error.
+
+Refer to all available `bb` commands linked above for full list of functionality.
+
+##### Generating Solidity verifier
+
+1. Follow [the Noir docs](https://noir-lang.org/docs/getting_started/hello_noir/) to compile and generate witness of your Noir program
+
+2. Prove the valid execution of your Noir program running:
+
+    ```bash
+    bb prove_keccak_ultra_honk -b ./target/hello_world.json -w ./target/witness-name.gz -o ./target/proof
```

Comment:
This line is different than what is shown above. Should it be this?

```suggestion
    bb prove_ultra_honk -b ./target/hello_world.json -w ./target/witness-name.gz -o ./target/proof
```

Reply:
And is this step required to generate the Solidity verifier? Or do we just need to create the verification key and generate the solidity verifier from that?

Reply:
Proofs need to be the Keccak-based version (instead of Poseidon-based) in order to be verifiable with the Solidity verifier is what I was told.

Reply:
ok i see. if that is the case I see why including this example of generating a proof is valuable, but the title of the section being "Generating Solidity Verifier" is confusing. This is showing how to generate a proof to be used with a Solidity verifier as well as the commands to actually generate the verifier. I think it would be helpful to clarify to add a note that `prove_keccak_ultra_honk` should be used for generating proofs to be verified by a solidity verifier. It's not currently clear i dont think

Reply:
Good advice.
Revised section title and supplemented note in the latest commit, see if it's better?